File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    var ans = 0, mn = prices[0];
    for(let i=0; i<prices.length; i++){
        mn = Math.min(mn, prices[i]);
        ans = Math.max(ans, prices[i]-mn);
    }
    return ans;
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mn = Integer.MAX_VALUE, ans = 0;
        for(int i=0; i<prices.length; i++){
            mn = Math.min(mn, prices[i]);
            ans = Math.max(ans, prices[i] - mn);
        }
        return ans;
    }
}